### PR TITLE
Installs git and implements basic github repo clone 

### PR DESCRIPTION
**Pull request by scriptnull**
#252 
#253 

This PR only supports a basic workflow of cloning github repositories in a runSh job
image

Things remaining ( to be done in future PRs)
- Add Pull request support
- Complete script for providers other than github
- Support private repositories

### CODE REVIEW

File: job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  {
+    $DateTime = (Get-Date).ToUniversalTime()
+    $cmd_end_timestamp = [System.Math]::Truncate((Get-Date -Date $DateTime -UFormat %s))
+    #$cmd_end_timestamp = Get-Date -format "%s"
+    Write-Output ""
+    Write-Output "__SH__CMD__END__|{`"type`":`"cmd`",`"sequenceNumber`":`"$cmd_end_timestamp`",`"id`":`"$cmd_uuid`",`"exitcode`":`"$cmd_status`"}|$cmd"
+  }
+}
+
+$PRIVATE_KEY = "<%=privateKey%>"
```
Comment (rageshkrishna) [inline job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1]: Shouldn't this use `@'`?

File: job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+$PROJECT_CLONE_URL = "<%=projectUrl%>"
+$PROJECT_CLONE_LOCATION = "<%=cloneLocation%>"
+$COMMIT_SHA = "<%=commitSha%>"
+$IS_PULL_REQUEST = <%= shaData.isPullRequest ? "$TRUE" : "$FALSE" %>
+$IS_PULL_REQUEST_CLOSE = <%= shaData.isPullRequestClose ? "$TRUE" : "$FALSE" %>
+$PULL_REQUEST = "<%=shaData.pullRequestNumber%>"
+$PULL_REQUEST_BASE_BRANCH = "<%=shaData.pullRequestBaseBranch%>"
```
Comment (rageshkrishna) [inline job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1]: Maybe this, too? Can branch names have `"`?

**Comment by scriptnull** [comment]
Updated `@'` in all places, as it gives us the nice guarantee of escaping characters.

File: job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1 [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+$PULL_REQUEST_BASE_BRANCH = @'
+<%=shaData.pullRequestBaseBranch%>
+'@
+$PROJECT = @'
+<%=name%>
+'@
+
+
+Function git_sync() {
+  $temp_clone_path = "C:\Users\ContainerAdministrator\Shippable\gitRepo"
```
Comment (rageshkrishna) [inline job/handlers/resources/WindowsServer_2016/gitRepo/templates/providers/_github.ps1]: Sorry, I missed this last time. No real reason to hard code ContainerAdministrator here. `Join-Path "$env:TEMP" "Shippable\gitRepo"` might be better.

**Pull request merged**